### ISSUE TITLE
storage resize_ function

### ISSUE THREAD

**Issue by mingminzhen** [gpleiss/efficient_densenet_pytorch#26]
I try to use torch.storage in my network. I use Pytorch3.

```
if self.storage.size() < size:
            is_cuda = self.storage.is_cuda
            if is_cuda:
                gpu_ID = self.storage.get_device()
                print('gpu_ID1:',gpu_ID)
            self.storage.resize_(size)
            
            gpu_ID= self.storage.get_device()
            print('gpu_ID2:',gpu_ID)
            
            if is_cuda:
                self.storage = self.storage.cuda(gpu_ID)
            gpu_ID= self.storage.get_device()
            print('gpu_ID3:',gpu_ID)
```

The output is 
```
gpu_ID1: 1
gpu_ID2: 0
gpu_ID3: 0

```
The self.storage comes from ``` self.storage= torch.Storage(1024)```.
It seems the **resize_** function will change the gpu where the storage will be saved.
I wish the storage is saved in GPU 1 rather than 0. 
How can i do that?

**Comment by gpleiss** [comment]
Right now things are a bit broken on PyTorch 0.3. I'm working on a fix for it.

**Comment by mingminzhen** [comment]
In pytorch0.3, "with" can solve this problem.

```
    def resize_(self, size):
        if self.storage.size() < size:
            is_cuda = self.storage.is_cuda
            if is_cuda:
                gpu_ID = self.storage.get_device()
                with torch.cuda.device(gpu_ID):
                    self.storage.resize_(size)
            else:
                self.storage.resize_(size)
        return self
```

**Comment by gpleiss** [comment]
Thanks @mingminzhen ! I'm going to push some 0.3 fixes soon, so I'll include that.

**Comment by gpleiss** [comment]
This should be fixed with #28.
